src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -4051,16 +4051,7 @@ void AuraEffect::HandleAuraModIncreaseEnergyPercent(AuraApplication const* aurAp
     UnitMods unitMod = UnitMods(UNIT_MOD_POWER_START + powerType);
     float amount = float(GetAmount());
 
-    if (apply)
-    {
-        target->HandleStatModifier(unitMod, TOTAL_PCT, amount, apply);
-        target->ModifyPowerPct(powerType, amount, apply);
-    }
-    else
-    {
-        target->ModifyPowerPct(powerType, amount, apply);
-        target->HandleStatModifier(unitMod, TOTAL_PCT, amount, apply);
-    }
```

Comment:
why is this removed?

Reply:
In case unit does not allow modifying stats (Unit::CanModifyStats()), we shouldn't force modify power here (I see in some player cases modify stats is set to false). I tested aura application/removal (for +pct mana and energy) and it works just fine, except if I'm missing something - not sure what the idea was here except to get the modified amount, which isn't used anyway.

Reply:
when you gain this aura, current power should be increased too, on remove you lose the power again

Reply:
Not if CanModifyStats is set to false (it makes sense that it should work like that), check Unit:9207, HandleStatModifier already updates max power. That's why it works exactly the same like before, this ugly code is not needed here :P

Reply:
i'm not talking about max power, current power is affected

Reply:
if you decrease max power, (current) power is set to max power (Unit.cpp, line 9513); if you increase max power, (current) power remains the same - expected behavior, no?

Reply:
following that code, no

Reply:
changing currect power here makes no sense, we should not give power, it should regenerate

Reply:
example https://wow.gamepedia.com/Hymn_of_Hope

Reply:
you are right, I only looked at 7.x dbc 😞 i'll revert this

Reply:
rewritten tho, reducing for pct of max power that is already increased by pct is not right 😀